src/test/java/org/broadinstitute/hellbender/utils/iterators/IntervalOverlappingIteratorUnitTest.java

```diff
@@ -59,13 +57,15 @@
     @Test(dataProvider = "data")
     public void testIterator(List<SimpleInterval> intervals, SAMSequenceDictionary dictionary, Locatable[] records,	Locatable[] expected) {
         IntervalOverlappingIterator<Locatable> iterator = new IntervalOverlappingIterator<>(Arrays.asList(records).iterator(), intervals, dictionary);
-        int index = 0;
-        final Iterable<Locatable> iterable = () -> iterator;
-        for(Locatable loc: iterable) {
+
```

Comment:
Ok I see why it is more difficult, in truth this is just testing code and I normally I just don't care much about it unless the coverage is bad...

I guess it my review was an automatics response when I saw again the iterator to iterable transformation trick. 

I would say that it is sufficient to add al the elements returned by the iterator and compare the corresponding list directly with the expected list without bothering to transform the expected list into an iterator and compare them that way.... up to you, I will accept this as it is since, again, I don't care about test code that much.